featuretools/primitives/standard/transform_primitive.py

```diff
@@ -623,10 +631,108 @@ class NumericLag(TransformPrimitive):
+
+    Args:
+        periods (int): The number of periods by which to shift the input.
+            Default is 1. Periods correspond to rows.
+
+        fill_value (int, float, str, bool, optional): The value to use to fill in
+            the gaps left after shifting the input. Default is None.
+
+    Examples:
+        >>> lag = Lag()
+        >>> lag([1, 2, 3, 4, 5], pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D'))).tolist()
+        [nan, 1.0, 2.0, 3.0, 4.0]
+
+        You can specify the number of periods to shift the values
+
+        >>> lag_periods = Lag(periods=3)
+        >>> lag_periods(["hello", "world", "test", "foo", "bar"], pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D'))).tolist()
+        [nan, nan, nan, 'hello', 'world']
+
+        You can specify the fill value to use
+
+        >>> lag_fill_value = Lag(fill_value=100)
+        >>> lag_fill_value([1, 2, 3, 4], pd.Series(pd.date_range(start="2020-01-01", periods=4, freq='D'))).tolist()
+        [100, 1, 2, 3]
+    """
+
+    name = "lag"
+    input_types = [
+        [
+            ColumnSchema(semantic_tags={"category"}),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=AgeFractional),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=AgeNullable),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=BooleanNullable),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=Double),
+            ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=EmailAddress),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=IntegerNullable),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=NaturalLanguage),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=Ordinal),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=PostalCode),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=URL),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=Unknown),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+    ]
+    return_type = None
+    uses_full_dataframe = True
+
+    def __init__(self, periods=1, fill_value=None):
+        self.periods = periods
+        self.fill_value = fill_value
+
+    def get_function(self):
+        def lag(input_col, time_index):
```

Comment:
I don't think this matters much, but we have tended to pass the time index as the first argument for primitives that use it along with another data column

Reply:
This was done because of the way `return_type=None` works. It grabs the first input type so it was assuming `time_index` for the output if I did it that way.

Reply:
Got it. Didn't realize that is how the return type was getting set, but that makes sense.